portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -136,12 +134,21 @@ private String getParameter(HttpServletRequest request, String name, String defa
      * @return the request parameter as an int, or the default value if the parameter
      *         was not specified by the client
      */
-    private int getIntParameter(HttpServletRequest request, String userValue, int defaultValue) {
+    private int getIntParameter(HttpServletRequest request, ArrayList<Integer> errors, String userValue, int defaultValue) throws IOException {
         String toParse = request.getParameter(userValue);
         if ("".equals(toParse) || toParse == null) {
             return defaultValue;
         }
-        int value = Integer.parseInt(toParse);
+        int value = defaultValue;
+        try{ 
+            value = Integer.parseInt(toParse);
+        } catch (NumberFormatException nfe) {
+            errors.add(400);
+            return defaultValue;
+        } catch (Exception e){
+            errors.add(500);
+            return defaultValue;
+        }    
         if (value < 0) {
```

Comment:
What if the default value < 0?

Reply:
Hmm, I'd probably handle this by returning 0.